common/djangoapps/student/tests/test_enrollment.py

```diff
@@ -281,3 +282,18 @@ def _change_enrollment(self, action, course_id=None, email_opt_in=None):
             params['email_opt_in'] = email_opt_in
 
         return self.client.post(reverse('change_enrollment'), params)
+
+    def test_get_or_create_integrity_error(self):
+        """Verify that get_or_create_enrollment handles IntegrityError."""
+
+        CourseEnrollment.objects.create(course_id=self.course.id,user=self.user)
+
+        from nose.tools import set_trace; set_trace()
+
+        with patch.object(CourseEnrollment.objects, "get_or_create") as mock_get_or_create:
+            mock_get_or_create.side_effect = IntegrityError
+            with self.assertRaises(IntegrityError):
```

Comment:
Isn't the expected behavior here that you don't raise an `IntegrityError`? (That if `get_or_create`, raises one, then you commit and then try a `get`?)